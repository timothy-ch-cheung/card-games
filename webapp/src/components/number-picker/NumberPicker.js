import React from 'react';
import InputGroup from "react-bootstrap/InputGroup";
import FormControl from "react-bootstrap/FormControl";
import Button from "react-bootstrap/Button";

function NumberPicker(props) {
    const largeFont = {fontSize: "1.5rem", padding: "0rem 0.75rem", width: "40px"}

    return (
        <InputGroup style={{width: "150px"}}>
            <InputGroup.Prepend>
                <Button variant="info" style={largeFont} onClick={props.onDecrease}
                        data-test={props.name + "-minus-btn"}>-</Button>
            </InputGroup.Prepend>
            <FormControl className="number-picker" disabled name={props.name}
                         style={{backgroundColor: "#fff", textAlign: "center"}} value={props.value}
                         data-test={props.name + "-value"}/>
            <InputGroup.Append>
                <Button variant="info" style={largeFont} onClick={props.onIncrease}
                        data-test={props.name + "-plus-btn"}>+</Button>
            </InputGroup.Append>
        </InputGroup>
    );
}

export default NumberPicker;