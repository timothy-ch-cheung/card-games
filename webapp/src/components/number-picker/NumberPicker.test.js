import React from "react";
import NumberPicker from "./NumberPicker";
import {configure, shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({adapter: new Adapter()});

describe("TEST SUITE NumberPicker: ", () => {
    let wrapper;
    let increase;
    let decrease;

    beforeEach(() => {
        increase = jest.fn();
        decrease = jest.fn();
        wrapper = shallow(<NumberPicker value={123} name={"number"} onIncrease={increase} onDecrease={decrease}/>);
    });

    test('Matches PlayerList snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('onDecrease is called when minus button is clicked', () => {
        wrapper.find('[data-test="number-minus-btn"]').simulate("click");
        expect(decrease).toHaveBeenCalledTimes(1);
        expect(increase).toHaveBeenCalledTimes(0);
    });

    test('onIncrease is called when plus button is clicked', () => {
        wrapper.find('[data-test="number-plus-btn"]').simulate("click");
        expect(increase).toHaveBeenCalledTimes(1);
        expect(decrease).toHaveBeenCalledTimes(0);
    });
});