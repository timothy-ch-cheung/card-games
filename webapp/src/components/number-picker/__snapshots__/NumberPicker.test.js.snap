// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST SUITE NumberPicker:  Matches PlayerList snapshot 1`] = `
<InputGroup
  style={
    Object {
      "width": "150px",
    }
  }
>
  <InputGroupPrepend>
    <Button
      active={false}
      data-test="number-minus-btn"
      disabled={false}
      onClick={[MockFunction]}
      style={
        Object {
          "fontSize": "1.5rem",
          "padding": "0rem 0.75rem",
          "width": "40px",
        }
      }
      type="button"
      variant="info"
    >
      -
    </Button>
  </InputGroupPrepend>
  <FormControl
    className="number-picker"
    data-test="number-value"
    disabled={true}
    name="number"
    style={
      Object {
        "backgroundColor": "#fff",
        "textAlign": "center",
      }
    }
    value={123}
  />
  <InputGroupAppend>
    <Button
      active={false}
      data-test="number-plus-btn"
      disabled={false}
      onClick={[MockFunction]}
      style={
        Object {
          "fontSize": "1.5rem",
          "padding": "0rem 0.75rem",
          "width": "40px",
        }
      }
      type="button"
      variant="info"
    >
      +
    </Button>
  </InputGroupAppend>
</InputGroup>
`;
